frontend/src/Pages/StartLesson.tsx

```diff
@@ -1,33 +1,124 @@
-import { Fragment } from "react";
+import { Fragment, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import { useAppSelector, useAppDispatch } from "../app/hooks";
+
+import {
+  fetchQuizDetailAsync,
+  fetchQuestionsAsync,
+  selectLesson,
+} from "../features/StartLesson/lessonSlice";
+import { Answer } from "../Types/Answer";
 
 import classes from "./StartLesson.module.css";
 
 const Lesson = () => {
+  const [currentQuestion, setCurrentQuestion] = useState(0);
+  const [showScore, setShowScore] = useState(false);
+  const [score, setScore] = useState(0);
+  const [answers, setAnswers] = useState<Answer[]>([]);
+
+  const params: { quizId: string } = useParams();
+
+  const { quizId } = params;
```

Comment:
You may simplify this further with 
```suggestion
  const { quizId } = useParams<quizId: string>();
```